substrate/frame/distribution/src/lib.rs

```diff
@@ -0,0 +1,216 @@
+	pub(super) type Spends<T: Config> =
+		StorageMap<_, Twox64Concat, SpendIndex, SpendInfo<T>, OptionQuery>;
+
+	/// List of whitelisted projects to be rewarded
+	#[pallet::storage]
+	pub type Projects<T: Config> =
+		StorageValue<_, BoundedVec<ProjectInfo<T>, T::MaxProjects>, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Reward successfully claimed
+		RewardClaimed {
+			when: BlockNumberFor<T>,
+			amount: BalanceOf<T>,
+			project_account: ProjectId<T>,
+		},
+
+		/// A Spend was created
+		SpendCreated {
+			when: BlockNumberFor<T>,
+			amount: BalanceOf<T>,
+			project_account: ProjectId<T>,
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Not enough Funds in the Pot
+		InsufficientPotReserves,
+		/// The funds transfer operation failed
+		TransferFailed,
+		/// Spend or Spend index does not exists
+		InexistentSpend,
+		/// No valid Account_id found
+		NoValidAccount,
+		/// No project available for funding
+		NoProjectAvailable,
+		/// The Funds transfer failed
+		FailedSpendOperation,
+		/// Still not in claiming period
+		NotClaimingPeriod,
+		/// Funds locking failed
+		FundsReserveFailed,
+		/// An invalid result  was returned
+		InvalidResult,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Weight: see `begin_block`
+		fn on_initialize(n: BlockNumberFor<T>) -> Weight {
+			Self::begin_block(n)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// OPF Reward Claim logic
+		///
+		/// ## Dispatch Origin
+		///
+		/// Must be signed
+		///
+		/// ## Details
+		///
+		/// From this extrinsic any user can claim a reward for a nominated/whitelisted project.
+		///
+		/// ### Parameters
+		/// - `project_account`: The account that will receive the reward.
+		///
+		/// ### Errors
+		/// - [`Error::<T>::InexistentSpend`]:Spend or Spend index does not exists
+		/// - [`Error::<T>::NoValidAccount`]:  No valid Account_id found
+		/// - [`Error::<T>::NotClaimingPeriod`]: Still not in claiming period
+		///  
+		/// ## Events
+		/// Emits [`Event::<T>::RewardClaimed`] if successful for a positive approval.
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::claim_reward_for(T::MaxProjects::get()))]
+		pub fn claim_reward_for(
+			origin: OriginFor<T>,
+			project_account: ProjectId<T>,
+		) -> DispatchResult {
+			let _caller = ensure_signed(origin)?;
+			let spend_indexes = Self::get_spend(project_account);
+			let pot = Self::pot_account();
+			for i in spend_indexes {
+				let info = Spends::<T>::get(i).ok_or(Error::<T>::InexistentSpend)?;
+				let project_account =
+					info.whitelisted_project.clone().ok_or(Error::<T>::NoValidAccount)?;
+				let now = T::BlockNumberProvider::current_block_number();
+
+				// Check that we're within the claiming period
+				ensure!(now > info.valid_from, Error::<T>::NotClaimingPeriod);
+				// Unlock the funds
+				T::NativeBalance::release(
```

Comment:
What is the purpose of holding the funds in the pot account? 

The pot account has no private key anyhow, so no one has access to it.

Reply:
It's helpful to track the pot transactions.